compiler/src/iree/compiler/Dialect/LinalgExt/Transforms/DecomposeWinogradPass.cpp

```diff
@@ -0,0 +1,276 @@
+  using OpRewritePattern<TransformOp>::OpRewritePattern;
+
+  LogicalResult matchAndRewrite(TransformOp transformOp,
+                                PatternRewriter &rewriter) const override {
+    auto originalType =
+        dyn_cast<RankedTensorType>(transformOp.getOriginalOperandType());
+    auto transformedType =
+        dyn_cast<RankedTensorType>(transformOp.getTransformedOperandType());
+    if (!originalType || !transformedType) {
+      return failure();
+    }
+    auto originalShape = originalType.getShape();
+    auto transformedShape = transformedType.getShape();
+    auto hwDims = transformOp.hwDimensions();
+    SetVector<int64_t> hwDimsSet(hwDims.begin(), hwDims.end());
+    if (!llvm::all_of(llvm::enumerate(originalShape),
+                      [&](auto it) {
+                        return it.value() == 1 ||
+                               hwDimsSet.contains(it.index());
+                      }) ||
+        originalShape.size() == 2) {
+      return failure();
+    }
+    Location loc = transformOp->getLoc();
+    SmallVector<int64_t> newOriginalShape = llvm::map_to_vector(
+        hwDims, [&](int64_t dim) { return originalShape[dim]; });
+    auto newOriginalType = originalType.clone(newOriginalShape);
+    SmallVector<int64_t> newTransformedShape(transformedShape.begin(),
+                                             transformedShape.begin() + 2);
+    auto newTransformedType = transformedType.clone(newTransformedShape);
+    RankedTensorType newInputType = newOriginalType;
+    RankedTensorType newOutputType = newTransformedType;
+    if (std::is_same_v<TransformOp, WinogradOutputTransformOp>) {
+      newInputType = newTransformedType;
+      newOutputType = newOriginalType;
+    }
+
+    auto newInput = tensor::createCanonicalRankReducingExtractSliceOp(
+        rewriter, loc, transformOp.getInputs()[0], newInputType);
+    auto newInit = tensor::createCanonicalRankReducingExtractSliceOp(
+        rewriter, loc, transformOp.getOutputs()[0], newOutputType);
+    auto rankReducedTransform = clone(rewriter, transformOp, newOutputType,
+                                      ValueRange{newInput, newInit});
+    auto insertSliceOp = tensor::createCanonicalRankReducingInsertSliceOp(
+        rewriter, loc, rankReducedTransform->getResult(0),
+        transformOp.getOutputs()[0]);
+    rewriter.replaceOp(transformOp, insertSliceOp);
+
+    return success();
+  }
+};
+
+/// Pattern to decompose the tiled WinogradInputTransformOp.
+struct DecomposeWinogradInputTransform
+    : public OpRewritePattern<WinogradInputTransformOp> {
+  using OpRewritePattern<WinogradInputTransformOp>::OpRewritePattern;
+
+  LogicalResult matchAndRewrite(WinogradInputTransformOp transformOp,
+                                PatternRewriter &rewriter) const override {
+    Location loc = transformOp.getLoc();
+    auto funcOp = transformOp->getParentOfType<mlir::FunctionOpInterface>();
```

Comment:
Hmmm, this is strange... this would only work if winograd ops are by themselves in a dispatch. Do we need this?

Reply:
I'm not sure exactly what this is for. I didn't mess with the decomposition, so maybe @harsh-nod could provide some context.

Reply:
Actually it seems that maybe the point is to create the constant tensors outside of the tiling loops? It should get hoisted out though, so I'm not sure. Let me see what happens when I remove it.